sources/api/sundog/src/main.rs

```diff
@@ -189,8 +194,33 @@ where
         }
     );
 
-    let generators: HashMap<String, String> = serde_json::from_str(&response_body)
+    let response: HashMap<String, Value> = serde_json::from_str(&response_body)
         .context(error::ResponseJsonSnafu { method: "GET", uri })?;
+
+    get_setting_generator_from_api_response(response)
+}
+
+fn get_setting_generator_from_api_response(
```

Comment:
I'm OK with this as-is, but there's [an example in the `serde` docs for parsing a struct from input which could either be a struct or a string](https://serde.rs/string-or-struct.html).

If implemented this way, you could leave the details of parsing settings generators to the model, and then just directly parse into `HashMap<String, SettingsGenerator>`.